test/over_react/util/react_wrappers_test.dart

```diff
@@ -574,6 +576,264 @@ main() {
+                'dartProp': 'dart',
+                'style': testStyle,
+                'children': testChildren
+              }));
+            });
+          });
+
+          group('Dart component ReactComponent', () {
+            test('', () {
+              ReactComponent renderedInstance = render(OneLevelWrapper()(
+                TestComponentFactory({
+                  'jsProp': 'js',
+                  'style': testStyle,
+                }, testChildren)
+              ));
+
+              expect(getProps(renderedInstance, traverseWrappers: true), equals({
+                'jsProp': 'js',
+                'style': testStyle,
+                'children': testChildren
+              }));
+            });
+
+            test('even when there are multiple levels of wrappers', () {
+              ReactComponent renderedInstance = render(TwoLevelWrapper()(
+                OneLevelWrapper()(
+                  TestComponentFactory({
+                    'jsProp': 'js',
+                    'style': testStyle,
+                  }, testChildren)
+                )
+              ));
+
+              expect(getProps(renderedInstance, traverseWrappers: true), equals({
+                'jsProp': 'js',
+                'style': testStyle,
+                'children': testChildren
+              }));
+            });
+
+            test('even when props change', () {
+              var mountNode = new DivElement();
+              ReactComponent renderedInstance = react_dom.render(OneLevelWrapper()(
+                TestComponentFactory({
+                  'jsProp': 'js',
+                  'style': testStyle,
+                }, testChildren)
+              ), mountNode);
+
+              expect(getProps(renderedInstance, traverseWrappers: true), equals({
+                'jsProp': 'js',
+                'style': testStyle,
+                'children': testChildren
+              }));
+
+              renderedInstance = react_dom.render(OneLevelWrapper()(
+                TestComponentFactory({
+                  'jsProp': 'other js',
+                  'style': testStyle,
+                }, testChildren)
+              ), mountNode);
+
+              expect(getProps(renderedInstance, traverseWrappers: true), equals({
+                'jsProp': 'other js',
+                'style': testStyle,
+                'children': testChildren
+              }));
+            });
+
+            test('except when traverseWrappers is false', () {
+              ReactComponent renderedInstance = render(OneLevelWrapper()(
+                TestComponentFactory({
+                  'jsProp': 'js',
+                  'style': testStyle,
+                }, testChildren)
+              ));
+
+              expect(getProps(renderedInstance), isNot({
+                'jsProp': 'other js',
+                'style': testStyle,
+                'children': testChildren
+              }));
+            });
+          });
+        });
+      });
+
```

Comment:
It'd probably be good to also test that `getProps` w/ `traverseWrappers: true` doesn't have any issues when passed a non-wrapper component.